appium/webdriver/common/touch_action.py

```diff
@@ -24,16 +24,23 @@
 # pylint: disable=no-self-use
 
 import copy
+from typing import Dict, List, TypeVar
 
 from appium.webdriver.mobilecommand import MobileCommand as Command
 
+# from appium.webdriver.webelement import WebElement as WebElement
+# from appium.webdriver.webdriver import WebDriver as WebDriver
+
+T = TypeVar('T', bound='TouchAction')
+
 
 class TouchAction(object):
+
     def __init__(self, driver=None):
         self._driver = driver
-        self._actions = []
+        self._actions: List = []
 
-    def tap(self, element=None, x=None, y=None, count=1):
+    def tap(self: T, element=None, x: int = None, y: int = None, count: int = 1) -> T:
```

Comment:
not sure if we need to also set the type of `self` arg

Reply:
It's necessary.

https://mypy.readthedocs.io/en/stable/generics.html#generic-methods-and-generic-self

Reply:
makes sense